core/src/main/java/org/neo4j/cypher/export/CypherDatabaseSubGraph.java

```diff
@@ -0,0 +1,223 @@
+                .distinct()
+                .collect( Collectors.toList());
+        this.relationshipTypes = StreamSupport.stream( this.relationships.spliterator(), false )
+                .map(relationship ->  relationship.getType())
+                .distinct()
+                .collect( Collectors.toList());
+    }
+
+    public static SubGraph from( Transaction transaction, String query )
+    {
+        return new CypherDatabaseSubGraph( transaction, query );
+    }
+
+    @Override
+    public Iterable<Node> getNodes()
+    {
+        return this.nodes;
+    }
+
+    @Override
+    public Iterable<Relationship> getRelationships(){
+        return this.relationships;
+    }
+
+    @Override
+    public boolean contains( Relationship relationship )
+    {
+        return transaction.getRelationshipById( relationship.getId() ) != null;
+    }
+
+    @Override
+    public Iterable<IndexDefinition> getIndexes()
+    {
+        Comparator<IndexDefinition> comp = Comparator.comparing(IndexDefinition::getName);
+        return Stream.concat(
+                        StreamSupport.stream(getAllLabelsInUse().spliterator(), false)
+                                .flatMap(label -> StreamSupport.stream(getIndexes(label).spliterator(), false))
+                        ,
+                        StreamSupport.stream(getAllRelationshipTypesInUse().spliterator(), false)
+                                .flatMap(relType -> StreamSupport.stream(getIndexes(relType).spliterator(), false))
+                ).sorted(comp)
+                .collect( Collectors.toList());
+    }
+
+    @Override
+    public Iterable<ConstraintDefinition> getConstraints()
+    {
+        Comparator<ConstraintDefinition> comp = Comparator.comparing(ConstraintDefinition::getName);
+        return Stream.concat(
+                        StreamSupport.stream(getAllLabelsInUse().spliterator(), false)
+                                .flatMap(label -> StreamSupport.stream(getConstraints(label).spliterator(), false))
+                        ,
+                        StreamSupport.stream(getAllRelationshipTypesInUse().spliterator(), false)
+                                .flatMap(relType -> StreamSupport.stream(getConstraints(relType).spliterator(), false))
+                ).sorted(comp)
+                .collect( Collectors.toList());
+    }
+
+    @Override
+    public Iterable<ConstraintDefinition> getConstraints(Label label) {
+        return transaction.schema().getConstraints(label);
+    }
+
+    @Override
+    public Iterable<ConstraintDefinition> getConstraints(RelationshipType type) {
+        return transaction.schema().getConstraints(type);
+    }
+
+    @Override
+    public Iterable<IndexDefinition> getIndexes(Label label) {
+        return transaction.schema().getIndexes(label);
+    }
+
+    @Override
+    public Iterable<IndexDefinition> getIndexes(RelationshipType type) {
+        return transaction.schema().getIndexes(type);
+    }
```

Comment:
these methods return the real information even if the label or the relationship type is not in the subgraph.